### PR TITLE
Routing setup

### LINKED ISSUE #3: Module 2 – Setting Up Routing
- Set up the main routes: Home, Sign In, Sign Up, Article, Profile, and Settings.
- Create page components for each route with basic layouts.

### PR DESCRIPTION
**Pull request by mikemakohon**
Resolves #3 
 - added routing for pages;
 - added pages boilerplates;
 - added basic header;

### CODE REVIEW

File: app/pages/ArticlePage/ArticlePage.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Delete article button (only shown to article’s author)
```
Comment (what1s1ove) [inline app/pages/ArticlePage/ArticlePage.tsx:1]: Name all components in the kebab case, `article-page` (ALL lowercase letters, words separated by `-`)

File: app/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Link } from "@remix-run/react";
```
Comment (what1s1ove) [inline app/components/Header/Header.tsx:1]: create your own `Link` component. It should be a wrapper for the Remix Link component

File: app/pages/ArticlePage/ArticlePage.tsx (hunk 1 above)
Comment (mikemakohon) [reply app/pages/ArticlePage/ArticlePage.tsx]: Changed, but for some reason github doesn't update it in the "files changed"
image

File: app/components/CustomLink/link.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Link } from "@remix-run/react";
+
+interface CustomLinkProps {
```
Comment (what1s1ove) [inline app/components/CustomLink/link.tsx:3]:
```suggestion
type Props {
```
simplify it

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**